File: typescript/src/models/address.ts
// this file is @generated

export interface Address {
  name: string;
  lineOne: string;
  lineTwo?: string | null;
  city: string;
  state: string;
  country: string;
  postalCode: string;
}

export const AddressSerializer = {
  _fromJsonObject(object: any): Address {
    return {
      name: object["name"],
      lineOne: object["line_one"],
      lineTwo: object["line_two"],
      city: object["city"],
      state: object["state"],
      country: object["country"],
      postalCode: object["postal_code"],
    };
  },

  _toJsonObject(self: Address): any {
    return {
      name: self.name,
      line_one: self.lineOne,
      line_two: self.lineTwo,
      city: self.city,
      state: self.state,
      country: self.country,
      postal_code: self.postalCode,
    };
  },
};

File: rust/src/models/address.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct Address {
    pub name: String,
    pub line_one: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub line_two: Option<String>,
    pub city: String,
    pub state: String,
    pub country: String,
    pub postal_code: String,
}
